packages/neo-one-smart-contract-lib/src/crowdsale/withAccessRoles/CrowdsaleWithCapperRole.ts

```diff
@@ -0,0 +1,47 @@
+import { Address, constant, MapStorage, createEventNotifier } from '@neo-one/smart-contract';
+import { AccessRoleHandler } from '../../utils';
+import { CrowdsaleContract } from '../CrowdsaleContract';
+
+const notifyAddCapper = createEventNotifier<Address, Address>('add_minter', 'address', 'requested by');
+const notifyRemoveCapper = createEventNotifier<Address, Address>('remove_minter', 'address', 'requested by');
+
+export abstract class CrowdsaleWithCapperRole extends CrowdsaleContract() {
+  private readonly capperList = MapStorage.for<Address, boolean>();
```

Comment:
Use `SetStorage` and eliminate the `AccessRoleHandler` entirely - it really doesn't save you that much code once you're using the right data structure.

Reply:
👍